tests/slurm_plugin/test_capacity_block_manager.py

```diff
@@ -0,0 +1,745 @@
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from datetime import datetime
+from unittest.mock import call
+
+import pytest
+from assertpy import assert_that
+from common.utils import SlurmCommandError
+from slurm_plugin.capacity_block_manager import (
+    SLURM_RESERVATION_NAME_PREFIX,
+    CapacityBlock,
+    CapacityBlockManager,
+    CapacityBlockManagerError,
+)
+from slurm_plugin.slurm_resources import DynamicNode, SlurmReservation, StaticNode
+
+from aws.common import AWSClientError
+from aws.ec2 import CapacityReservationInfo
+
+FAKE_CAPACITY_BLOCK_ID = "cr-a1234567"
+FAKE_CAPACITY_BLOCK_INFO = {
+    "CapacityReservationId": FAKE_CAPACITY_BLOCK_ID,
+    "EndDateType": "limited",
+    "ReservationType": "capacity-block",
+    "AvailabilityZone": "eu-east-2a",
+    "InstanceMatchCriteria": "targeted",
+    "EphemeralStorage": False,
+    "CreateDate": "2023-07-29T14:22:45Z  ",
+    "StartDate": "2023-08-15T12:00:00Z",
+    "EndDate": "2023-08-19T12:00:00Z",
+    "AvailableInstanceCount": 0,
+    "InstancePlatform": "Linux/UNIX",
+    "TotalInstanceCount": 16,
+    "State": "payment-pending",
+    "Tenancy": "default",
+    "EbsOptimized": True,
+    "InstanceType": "p5.48xlarge",
+}
+
+
+@pytest.fixture
+def capacity_block():
+    return CapacityBlock(FAKE_CAPACITY_BLOCK_ID, "queue-cb", "compute-resource-cb")
+
+
+class TestCapacityBlock:
+    @pytest.mark.parametrize(
+        ("state", "expected_output"),
+        [("active", True), ("anything-else", False)],
+    )
+    def test_is_active(self, capacity_block, state, expected_output):
+        capacity_block_reservation_info = CapacityReservationInfo({**FAKE_CAPACITY_BLOCK_INFO, "State": state})
+        capacity_block.update_capacity_block_reservation_info(capacity_block_reservation_info)
+        assert_that(capacity_block.is_active()).is_equal_to(expected_output)
+
+    @pytest.mark.parametrize(
+        "nodes_to_add",
+        [(["node1"]), (["node1", "node2"])],
+    )
+    def test_add_nodename(self, capacity_block, nodes_to_add):
+        for nodename in nodes_to_add:
+            capacity_block.add_nodename(nodename)
+        assert_that(capacity_block.nodenames()).is_equal_to(nodes_to_add)
+
+    @pytest.mark.parametrize(
+        ("node", "expected_output"),
+        [
+            (StaticNode("queue-cb-st-compute-resource-cb-4", "ip-1", "hostname-1", "some_state", "queue-cb"), True),
+            (StaticNode("queue1-st-c5xlarge-4", "ip-1", "hostname-1", "some_state", "queue1"), False),
+            (StaticNode("queue2-st-compute-resource1-4", "ip-1", "hostname-1", "some_state", "queue2"), False),
+        ],
+    )
+    def test_does_node_belong_to(self, capacity_block, node, expected_output):
```

Comment:
Please add test parameter cases where 
* node is in CB queue but not in CB compute resource
* node is not ni CB queue and in a compute resource with same name of CB

Reply:
Added.